### PR TITLE
extends history support to banis and ceremonies

### LINKED ISSUE #552: Banis should appear in history
- opening a Bani from SG should appear in history
- same with Anand Sahib (Bhog) in shortcut tray

Need a way to differentiate Banis vs Shabads in this history list. Maybe with a highlight color or a colored square next to the line. cc: @saintsoldierx

### PR DESCRIPTION
**Pull request by inderpreetsingh**
Adds history to support to banis, shabads and ceremonies. Reorganizes code for adding history. 

Fixes #552

### CODE REVIEW

File: www/js/viewer.js [hunk 1]
```diff
@@ -87,11 +87,13 @@ const castShabadLine = lineID => {
...
-    const activeSlide = document.querySelector('.deck.active .slide.active').children;
-    Array.prototype.forEach.call(activeSlide, element => {
-      const icons = iconsetHtml(`icons-${element.classList[0]}`, element.innerHTML);
-      if (icons) document.querySelector('.viewer-controls').appendChild(icons);
-    });
+    const activeSlide = document.querySelector('.deck.active .slide.active');
+    if (activeSlide) {
+      Array.prototype.forEach.call(activeSlide.children, element => {
+        const icons = iconsetHtml(`icons-${element.classList[0]}`, element.innerHTML);
+        if (icons) document.querySelector('.viewer-controls').appendChild(icons);
```
Comment (tsingh777) [inline www/js/viewer.js]: nit: add the curly braces for this `if` statement.

File: www/js/search.js [hunk 2]
```diff
@@ -708,6 +732,10 @@ module.exports = {
     banidb.loadBani(BaniID, baniLengthCols[baniLength]).then(rowsDb => {
       // create a unique shabadID for whole bani, and append it with length
       const shabadID = `${rowsDb[0].Token || rowsDb[0].Bani.Token}-${baniLength}`;
+      const nameOfBani = rowsDb[0].Gurmukhi || rowsDb[0].Bani.Gurmukhi;
```
Comment (tsingh777) [inline www/js/search.js]: Just wondering should the history for Anand sahib (Bhog) say: 
image
maybe we need to update db? to keep it consistent?

File: www/js/viewer.js (hunk 1 above)
Comment (inderpreetsingh) [reply www/js/viewer.js]: Oh that's pre-existing code I think. I will fix it.

File: www/js/search.js (hunk 2 above)
Comment (inderpreetsingh) [reply www/js/search.js]: @maneetpaul @saintsoldierx Please advice. _not a blocker_

**Comment by inderpreetsingh** [comment]
Let me add shabad status to it, and then we can merge together.

**Pull request merged**